### ISSUE TITLE
Typescript 3.1.3 import error gets stuck

### ISSUE THREAD

**Issue by jpike88** [Microsoft/vscode#60928]
- VSCode Version:1.28.1
- OS Version:MacOS Mojave

Steps to Reproduce:

1. Use typescript version 3.1.3 as workspace version
2. Cause an TS import error
3. It gets stuck

Does this issue occur when all extensions are disabled?: Yes
![screen shot 2018-10-15 at 10 47 44 am](https://user-images.githubusercontent.com/9585787/46929090-04c51080-d068-11e8-89cb-6ed1102d3b17.png)

**Comment by mjbvz** [comment]
Please provide more detailed repo steps. What is your tsconfig for example?

**Comment by jpike88** [comment]
```
{
  "compileOnSave": true,
	"compilerOptions": {
		"target": "esnext",
		"module": "esnext",
		"sourceMap": true,
		"moduleResolution": "node",
		"allowSyntheticDefaultImports": true,
		"noUnusedLocals": true,
		"resolveJsonModule": true,
		"noUnusedParameters": true,
		"noImplicitThis": true,
		"skipLibCheck": true,
		"outDir": "../dist/",
		"typeRoots": [
			"../node_modules/@types"
		],
		"baseUrl": "./",
		"paths": {
			"@shared/*": [
			  "../../shared/*"
			],
			"@seed/*": ["./components/*"],
			"@server/*": ["../../server/src/components/*"]
		}
	},
	"exclude": [
		"../node_modules/@types/"
	]
}
```

**Comment by jpike88** [comment]
Yes, it still persists if I switch to 3.1.1 (built in version to VSCode)

**Comment by jpike88** [comment]
This looks like it might be limited to imported JSONs using resolveJSONModule.

**Comment by jpike88** [comment]
@mjbvz I've renamed the issue to suit

**Comment by jpike88** [comment]
Nope, I was wrong on that, it's what it was before, purely related to import paths

**Comment by jpike88** [comment]
Ok even weirder behaviour. I'm going to paste a couple of video links to show you.

**Comment by jpike88** [comment]
- fresh boot
- video below happens.
- no other files are open
- happens with no extensions
- seems to only be triggered when changing the path of an imported JSON file.

http://streaka.testing.videos.s3.amazonaws.com/screencast_2018-10-16_11-55-01.mp4

**Comment by jpike88** [comment]
My other developers have this issue too, not just me. All same versions

**Comment by jpike88** [comment]
@mjbvz this issue is still 'needs more info'... is there anything else I should be providing?

**Comment by mjbvz** [comment]
Stack looks like https://github.com/Microsoft/TypeScript/issues/27405. Can you using `typescript@next` by following [these instructions](https://code.visualstudio.com/Docs/languages/typescript#_using-newer-typescript-versions) to see if that fixes the problem

I'm not sure if that error is the root cause of the issue you are seeing or not
